p5 with browserify gives "Error: Cannot find module '../core/core'" in v0.4.6 and v0.4.7

I seem to be running into errors when using p5js v0.4.6 and v0.4.7 with browserify. However, with v0.4.5, I have no issues. Browserify runs as expected. Any tips?

For example, if I run `browserify main.js -o bundle.js` on [this code](https://github.com/johnsojr/test-p5-browserify), I get these errors:

```
Error: Cannot find module '../core/core' from '/Users/jjohnson136/dev/p5-browserify-test/node_modules/p5/lib'
    at /Users/jjohnson136/.nvm/versions/node/v0.12.5/lib/node_modules/browserify/node_modules/resolve/lib/async.js:55:21
    at load (/Users/jjohnson136/.nvm/versions/node/v0.12.5/lib/node_modules/browserify/node_modules/resolve/lib/async.js:69:43)
    at onex (/Users/jjohnson136/.nvm/versions/node/v0.12.5/lib/node_modules/browserify/node_modules/resolve/lib/async.js:92:31)
    at /Users/jjohnson136/.nvm/versions/node/v0.12.5/lib/node_modules/browserify/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:95:15)
```

Thanks for any help!

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -58,7 +58,10 @@ module.exports = function(grunt) {
         reporter: require('jscs-stylish').path
       },
       build: {
-        src: ['Gruntfile.js']
+        src: [
+          'Gruntfile.js',
+          'build/**/*.js'
+        ]
       },
       source: {
         src: [
@@ -77,7 +80,10 @@ module.exports = function(grunt) {
         options: {
           jshintrc: '.jshintrc'
         },
-        src: ['Gruntfile.js']
+        src: [
+          'Gruntfile.js',
+          'build/**/*.js'
+        ]
       },
       source: {
         options: {

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -259,8 +259,10 @@ module.exports = function(grunt) {
     }
   });
 
+  // Load task definitions
+  grunt.loadTasks('build/tasks');
+
   // Load the external libraries used.
-  grunt.loadNpmTasks('grunt-browserify');
   grunt.loadNpmTasks('grunt-jscs');
   grunt.loadNpmTasks('grunt-contrib-jshint');
   grunt.loadNpmTasks('grunt-contrib-watch');

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -162,21 +162,6 @@ module.exports = function(grunt) {
       }
     },
 
-    // Build p5 into a single, UMD-wrapped file
-    browserify: {
-      p5: {
-        options: {
-          transform: ['brfs'],
-          browserifyOptions: {
-            standalone: 'p5'
-          },
-          banner: '/*! p5.js v<%= pkg.version %> <%= grunt.template.today("mmmm dd, yyyy") %> */'
-        },
-        src: 'src/app.js',
-        dest: 'lib/p5.js'
-      }
-    },
-
     // The actual compile step:  This should collect all the dependencies
     // and compile them into a single file.
     requirejs: {

--- a/build/tasks/browserify.js
+++ b/build/tasks/browserify.js
@@ -1,0 +1,44 @@
+'use strict';
+
+var path = require('path');
+var fs = require('fs');
+var browserify = require('browserify');
+var derequire = require('derequire'); // This is a silly thing to type
+
+var bannerTemplate = '/*! p5.js v<%= pkg.version %> <%= grunt.template.today("mmmm dd, yyyy") %> */';
+
+module.exports = function(grunt) {
+
+  var libFilePath = path.join(process.cwd(), 'lib/p5.js');
+  var srcFilePath = path.join(process.cwd(), 'src/app.js');
+
+  grunt.registerTask('browserify', 'Compile the p5.js source with Browserify', function() {
+    // Reading and writing files is asynchronous
+    var done = this.async();
+
+    // Render the banner for the top of the file
+    var banner = grunt.template.process(bannerTemplate);
+
+    // Invoke Browserify programatically to bundle the code
+    var bundle = browserify(srcFilePath, {
+        standalone: 'p5'
+      })
+      .transform('brfs')
+      .bundle();
+
+    // Start the generated output with the banner comment,
+    var code = banner + '\n';
+
+    // Then read the bundle into memory so we can run it through derequire
+    bundle.on('data', function(data) {
+      code += data;
+    }).on('end', function() {
+      // "code" is complete: create the distributable UMD build by running
+      // the bundle through derequire, then write the bundle to disk.
+      // (Derequire changes the bundle's internal "require" function to
+      // something that will not interfere with this module being used
+      // within a separate browserify bundle.)
+      fs.writeFile(libFilePath, derequire(code), done);
+    });
+  });
+};

--- a/build/tasks/browserify.js
+++ b/build/tasks/browserify.js
@@ -1,16 +1,17 @@
 'use strict';
 
 var path = require('path');
-var fs = require('fs');
 var browserify = require('browserify');
-var derequire = require('derequire'); // This is a silly thing to type
+var derequire = require('derequire');
 
 var bannerTemplate = '/*! p5.js v<%= pkg.version %> <%= grunt.template.today("mmmm dd, yyyy") %> */';
 
 module.exports = function(grunt) {
 
-  var libFilePath = path.join(process.cwd(), 'lib/p5.js');
-  var srcFilePath = path.join(process.cwd(), 'src/app.js');
+  var srcFilePath = require.resolve('../../src/app.js');
+
+  // This file will not exist until it has been built
+  var libFilePath = path.resolve('lib/p5.js');
 
   grunt.registerTask('browserify', 'Compile the p5.js source with Browserify', function() {
     // Reading and writing files is asynchronous
@@ -33,12 +34,19 @@ module.exports = function(grunt) {
     bundle.on('data', function(data) {
       code += data;
     }).on('end', function() {
+
       // "code" is complete: create the distributable UMD build by running
       // the bundle through derequire, then write the bundle to disk.
       // (Derequire changes the bundle's internal "require" function to
       // something that will not interfere with this module being used
       // within a separate browserify bundle.)
-      fs.writeFile(libFilePath, derequire(code), done);
+      grunt.file.write(libFilePath, derequire(code));
+
+      // Print a success message
+      grunt.log.writeln('>>'.green + ' Bundle ' + 'lib/p5.js'.cyan + ' created.');
+
+      // Complete the task
+      done();
     });
   });
 };

--- a/package.json
+++ b/package.json
@@ -13,9 +13,10 @@
   "version": "0.4.7",
   "devDependencies": {
     "amdclean": "~0.3.3",
+    "browserify": "^11.0.1",
     "concat-files": "^0.1.0",
+    "derequire": "^2.0.0",
     "grunt": "^0.4.5",
-    "grunt-browserify": "^3.8.0",
     "grunt-cli": "^0.1.13",
     "grunt-contrib-concat": "^0.5.1",
     "grunt-contrib-connect": "^0.9.0",

